src/main/java/com/mparticle/kits/ResponsysKit.java

```diff
@@ -0,0 +1,304 @@
+                                mPushIOManager.setPreference(entry.getKey(), entry.getValue());
+                            } catch (ValidationException e) {
+                                PIOLogger.v("RK lE ValidationException: " + e.getMessage());
+                            }
+                        }
+                    }
+                }
+                break;
+            case Other:
+                if (customFlags != null && customFlags.containsKey(CUSTOM_FLAG_IAM)) {
+                    mPushIOManager.trackEvent(mpEvent.getEventName());
+                } else {
+                    mPushIOManager.trackEngagement(PushIOManager.PUSHIO_ENGAGEMENT_METRIC_OTHER);
+                }
+                reportingMessages.add(ReportingMessage.fromEvent(this, mpEvent));
+                break;
+        }
+
+        return (reportingMessages.isEmpty() ? null : reportingMessages);
+    }
+
+    @Override
+    public List<ReportingMessage> logScreen(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public boolean willHandlePushMessage(Intent intent) {
+        PIOLogger.v("RK wHPM");
+        return isResponsysPush(intent);
+    }
+
+    @Override
+    public void onPushMessageReceived(Context context, Intent intent) {
+        PIOLogger.v("RK oPMR");
+        Intent newIntent = new Intent(intent);
+        newIntent.setClassName(context, PushIOGCMIntentService.class.getName());
+        JobIntentService.enqueueWork(context, PushIOGCMIntentService.class,
+                PushIOConstants.PIO_GCM_INTENT_SERVICE_JOB_ID,
+                newIntent);
+    }
+
+    @Override
+    public boolean onPushRegistration(String instanceId, String senderId) {
+        PIOLogger.v("RK oPR Instance ID: " + instanceId + ", Sender ID: " + senderId);
+        mPushIOManager.setDeviceToken(instanceId);
+        mPushIOManager.registerApp();
+        return true;
+    }
+
+    @Override
+    public void onIdentifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+    }
+
+    @Override
+    public void onLoginCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        PIOLogger.v("RK oLiC");
+        registerUserId(mParticleUser);
+    }
+
+    @Override
+    public void onLogoutCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        PIOLogger.v("RK oLoC");
+        if (mPushIOManager != null) {
+            mPushIOManager.unregisterUserId();
+        }
+    }
+
+    @Override
+    public void onModifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+    }
+
+    @Override
+    public void onUserIdentified(MParticleUser mParticleUser) {
+    }
+
+    private void registerUserId(MParticleUser mParticleUser) {
+        if (mPushIOManager != null) {
+            final String userId = getUserId(mParticleUser.getUserIdentities());
```

Comment:
not all customers provide email, esp if they're only looking to engage over push. In that case, they may only provide customer ID. A common practice for our integrations is to let the customers select which ID to use as the integration's user ID. We can add a setting like that if you think it's appropriate. Here's an example: https://github.com/mparticle-integrations/mparticle-android-integration-leanplum/blob/master/src/main/java/com/mparticle/kits/LeanplumKit.java#L119


Also - iOS uses customer ID rather than email. We need to make sure they're both the same.

Reply:
Switched to Customer ID: e4dd26c1702468166e1b3a55e717bf53a6e30637